tests/src/core/users/DetermineLifetimeConditions.test.ts

```diff
@@ -0,0 +1,214 @@
+  usersTiersRepository = testFactory.getUsersTiersRepository();
+  bit2MeService = new Bit2MeService(config, axios);
+  paymentService = new PaymentService(
+    new Stripe(config.STRIPE_SECRET_KEY, { apiVersion: '2024-04-10' }),
+    productsRepository,
+    bit2MeService,
+  );
+  usersService = new UsersService(
+    usersRepository,
+    paymentService,
+    displayBillingRepository,
+    couponsRepository,
+    usersCouponsRepository,
+    config,
+    axios,
+  );
+  storageService = new StorageService(config, axios);
+  tiersService = new TiersService(
+    usersService,
+    paymentService,
+    tiersRepository,
+    usersTiersRepository,
+    storageService,
+    config,
+  );
+
+  determineLifetimeConditions = new DetermineLifetimeConditions(paymentService, tiersService);
+});
+
+describe('Determining Lifetime conditions', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  describe('Handling stack lifetime', () => {
+    it('when there is no tier, then an error indicating so is thrown', async () => {
+      const user = getUser({ lifetime: true });
+      const customer = getCustomer({ id: user.customerId });
+      const tierNotFoundError = new TierNotFoundError(`Tier not found for user ${user.uuid} when stacking lifetime`);
+
+      jest.spyOn(paymentService, 'getCustomer').mockResolvedValue(customer as Stripe.Response<Stripe.Customer>);
+      jest.spyOn(paymentService, 'getCustomersByEmail').mockResolvedValue([customer]);
+      jest.spyOn(paymentService, 'getInvoicesFromUser').mockResolvedValue([]);
+      jest.spyOn(tiersService, 'getTiersProductsByUserId').mockRejectedValue(tierNotFoundError);
+      jest.spyOn(determineLifetimeConditions as any, 'getHigherTier').mockResolvedValue(null);
+
+      await expect((determineLifetimeConditions as any).handleStackingLifetime(user)).rejects.toThrow(
+        tierNotFoundError,
+      );
+    });
+
+    it('When we want to fetch the higher tier and the max space bytes, then the correct tier and bytes are returned', async () => {
+      const user = getUser({ lifetime: true });
+      const customer = getCustomer({ id: user.customerId });
+      const invoice = getInvoice();
+      const mockedTier = newTier({ billingType: 'lifetime' });
+
+      jest.spyOn(paymentService, 'getCustomer').mockResolvedValue(customer as Stripe.Response<Stripe.Customer>);
+      jest.spyOn(paymentService, 'getCustomersByEmail').mockResolvedValue([customer]);
+      jest.spyOn(paymentService, 'getInvoicesFromUser').mockResolvedValue([invoice]);
+      jest.spyOn(determineLifetimeConditions as any, 'getPaidInvoices').mockResolvedValue([invoice]);
+      jest.spyOn(tiersService, 'getTiersProductsByUserId').mockResolvedValue([mockedTier]);
+      jest.spyOn(determineLifetimeConditions as any, 'getHigherTier').mockResolvedValue(mockedTier);
+
+      const result = await (determineLifetimeConditions as any).handleStackingLifetime(user);
```

Comment:
Although using `as any` is discouraged, if due to the setup is currently needed, do it in the first line, assigning it to a variable and then use that variable on the rest of the lines